src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py

```diff
@@ -163,8 +164,11 @@ def __call__(self, wmspec, task, data=None, mask=None, team=None, continuous=Fal
             pileupDatasets = self.wmspec.listPileupDatasets()
             if pileupDatasets:
                 # unwrap {"url":[datasets]} structure into list of datasets
-                datasets = [d for prec in pileupDatasets.values() for d in prec]
-                self.pileupData = self.getDatasetLocations(datasets)
+                #datasets = [d for prec in pileupDatasets.values() for d in prec]
+                #self.pileupData = self.getDatasetLocations(datasets)
+                ### flattening the pileupDatasets to have (dbsurl, dataset) pairs to loop over in the getDatasetLocationsFromMSPileup
+                datasetsWithDbsURL = [(dbsUrl, dataset) for dbsUrl, datasets in pileupDatasets.items() for dataset in datasets]
```

Comment:
As I look into this, I would even say that we don't have to reorganize the data that comes out of `self.wmspec.listPileupDatasets()`. That structure would be in a format like:
```
{"cmsweb production DBSReader": [pileupA, pileupB, pileupC],
  "cmsweb-testbed DBSReader": [pileupD]}
```

Reply:
I've thought about this for a while, maybe for now it's better to keep it as it is. The alternative would be to have a for loop over `pileupDatasets`, and send to `getDatasetLocationsFromMSPileup` the key as the `dbsUrl` and the list `[pileupA, pileupB, pileupC]`. This would imply updating `self.pileupData` at each iteration of the outer loop. What do you think?